src/Symfony/Component/Messenger/Bridge/Amqp/Transport/AmqpReceiver.php

```diff
@@ -84,6 +85,10 @@ private function getEnvelope(string $queueName): iterable
             throw $exception;
         }
 
+        if (null !== $amqpEnvelope->getMessageId()) {
+            $envelope = $envelope->with(new TransportMessageIdStamp($amqpEnvelope->getMessageId()));
```

Comment:
to match what other receiver do:
```suggestion
            $envelope = $envelope
                ->withoutAll(TransportMessageIdStamp::class)
                ->with(new TransportMessageIdStamp($amqpEnvelope->getMessageId()));
```